src/main/java/com/shyb/evcharging/card/application/CardService.java

```diff
@@ -0,0 +1,46 @@
+package com.shyb.evcharging.card.application;
+
+import com.shyb.evcharging.card.domain.Card;
+import com.shyb.evcharging.card.dto.CardRequestDto;
+import com.shyb.evcharging.card.dto.CardResponseDto;
+import com.shyb.evcharging.card.repository.CardRepository;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CardService {
+
+    private final CardRepository cardRepository;
+
+    /**
+     * 카드를 등록한 후 리턴합니다.
+     *
+     * @param card 생성 요청 카드 정보
+     * @eturn 등록된 카드
+     */
+    public CardResponseDto save(CardRequestDto card) {
+
+        Card savedCard = cardRepository.save(card.toEntity());
+
+        return CardResponseDto.builder()
+            .id(savedCard.getId())
+            .cardNumber(savedCard.getCardNumber())
+            .cvc(savedCard.getCvc())
+            .expiryYear(savedCard.getExpiryYear())
+            .expiryMonth(savedCard.getExpiryMonth())
+            .build();
```

Comment:
서비스는 핵심 비즈니스 로직을 다루는 곳이에요.

만약에 builder 코드가 지저분하다고 생각되면 정적 팩토리 메소드(static)를 이용해 아래처럼 사용해보셔도 좋습니다.

```suggestion
        return CardResponseDto.from(savedCard);
```

Reply:
이펙티브 자바 아이템 1에 정적 팩토리 메소드와 관련된 부분이 나오는데

네이밍 규칙도 있으니 한번 아래 문서를 읽어보셔도 좋을 것 같네요

https://sun-22.tistory.com/84